lib/Reader/abisignature.cpp

```diff
@@ -150,14 +346,24 @@ Value *ABICallSignature::emitCall(GenIR &Reader, llvm::Value *Target,
   Type *FunctionPtrTy = Reader.getUnmanagedPointerType(FunctionTy);
 
   Target = Builder.CreateIntToPtr(Target, FunctionPtrTy);
-  CallSite Call = Reader.makeCall(Target, MayThrow, Arguments);
+
+  // Unmanaged calls require an intrinsic in order to ensure that any
+  // callee-save registers that might contain GC pointers are spilled.
```

Comment:
That's not the only reason for an intrinsic here, right? You are handling the GC mode switch and such there now too.